### PR TITLE
adds user input validations

### PR DESCRIPTION
**Pull request by uimarshall**
For your review

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -6,15 +6,26 @@ const router = express.Router();
 
 // Bring in the User Model
 const User = require('../../models/User');
+// Import Input Validation
+const validateRegInput = require('../../validation/register');
+// const validateLoginInput = require('../../validation/login');
```
Comment (enyeInc) [inline routes/api/users.js]: clean up code

File: routes/api/users.js [hunk 2]
```diff
@@ -6,15 +6,26 @@ const router = express.Router();
...
 
 // User Registration Route
 router.post('/register', async (req, res, next) => {
     try {
+        // Validate everything coming the request body
+        const { errors, isValid } = validateRegInput(req.body);
+        // Check Validation
+        if (!isValid) {
+            return res.status(400).json(errors);
+        }
```
Comment (enyeInc) [inline routes/api/users.js:21]: pad `if` statements with new lines

File: routes/api/users.js [hunk 3]
```diff
@@ -6,15 +6,26 @@ const router = express.Router();
...
         const { email, name, password } = req.body;
         // Check if the email coming in matches what is in the DB
         const user = await User.findOne({ email });
         if (user) {
-            return res.status(400).json({ message: `${email} already exist` });
+            errors.email = 'Email already exist';
+            return res.status(400).json(errors);
+            // return res.status(400).json({ message: `${email} already exist` });
```
Comment (enyeInc) [inline routes/api/users.js]: clean up code

File: validation/login.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const Validator = require('validator');
+const isEmpty = require('./is_empty');
+
+module.exports = function validateLoginInput(data) {
+    const errors = {};
+
+    // Validate Email
+    let checkemail = data.email;
```
Comment (enyeInc) [inline validation/login.js]: we use camel case in javascript. `checkEmail`

File: validation/login.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    checkemail = !isEmpty(checkemail) ? checkemail : '';
+    // Validate Password
+    let checkpassword = data.password;
```
Comment (enyeInc) [inline validation/login.js]: `checkPassword`

File: validation/register.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+const isEmpty = require('./is_empty');
+
+module.exports = function validateRegInput(data) {
+    const errors = {};
+    /* Check if data.name is not empty and return the name or Check that it is not null,
+    then validate it using the 'validator' which only validates string */
+
+    /* Assign 'data' pty in a func to variable to avoid Assignment
+    to property of function parameter 'data' no-param-reassign */
+    let checkname = data.name;
```
Comment (enyeInc) [inline validation/register.js]: fix all the camel casing errors on this file

File: routes/api/users.js (hunk 1 above)
Comment (uimarshall) [reply routes/api/users.js]: Done!

File: routes/api/users.js (hunk 2 above)
Comment (uimarshall) [reply routes/api/users.js:21]: Done!

File: routes/api/users.js (hunk 3 above)
Comment (uimarshall) [reply routes/api/users.js]: Done!

File: validation/login.js (hunk 4 above)
Comment (uimarshall) [reply validation/login.js]: Done!

File: validation/login.js (hunk 5 above)
Comment (uimarshall) [reply validation/login.js]: Done!

File: validation/register.js (hunk 6 above)
Comment (uimarshall) [reply validation/register.js]: Done!

File: routes/api/users.js (hunk 2 above)
Comment (enyeInc) [reply routes/api/users.js:21]: click `resolve conversation` once you have fixed the code

File: validation/login.js (hunk 5 above)
Comment (enyeInc) [reply validation/login.js]: is it?

File: validation/login.js (hunk 4 above)
Comment (enyeInc) [reply validation/login.js]: doesn't look like it

**Review by enyeInc — APPROVED** [review]

**Pull request merged**